packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -855,6 +900,21 @@ export class Combobox
           const item = this.filteredItems[this.activeItemIndex];
           this.toggleSelection(item, !item.selected);
           event.preventDefault();
+          if (this.selectAllEnabled) {
+            if (item === this.selectAllComboboxItemReferenceEl) {
```

Comment:
This logic seems the same as above. Can it share the same private method? See line 717.